### PR TITLE
googlecloud: renamed stackdriver exporter to googlecloud (1 of 2)

### PR DESCRIPTION
**Pull request by aabmass**
**Description:**
Renaming `stackdriver` exporter to `googlecloud`. This PR just moves all of the files and renames things where appropriate.

The next PR (https://github.com/aabmass/opentelemetry-collector-contrib/pull/47) adds back a stackdriver exporter (for compatibility with older configs) that minimally wraps this one and has a deprecation notice.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** Tested that the collector runs and exports successfully with a minimal config:

```yaml
receivers:
  hostmetrics:
    collection_interval: 10s
    scrapers:
      cpu:

exporters:
  googlecloud:
    project: otel-starter-project
    use_insecure: false
  logging:
    loglevel: debug
    sampling_initial: 200
    sampling_thereafter: 1

processors:
  batch:

service:
  pipelines:
    metrics:
      receivers: [hostmetrics]
      processors: [batch]
      exporters: [googlecloud, logging]
```

**Documentation:** 
Readme copied over. The next PR puts deprecation notices and migration instructions in the stackdriver exporter readme.

### CODE REVIEW

**Comment by aabmass** [comment]
@dashpole ptal as well :)

**Review by dashpole — APPROVED** [review]
lgtm!

**Comment by aabmass** [comment]
cc @punya

**Review by bogdandrutu — COMMENTED** [review]
Please run `make gendependabot`

**Comment by aabmass** [comment]
@bogdandrutu done

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,312 +1,82 @@
...
-      interval: "weekly"
-  - package-ecosystem: "gomod"
-    directory: "/testbed/mockdatareceivers/mockawsxrayreceiver"
-    schedule:
-      interval: "weekly"
-  - package-ecosystem: "gomod"
-    directory: "/tracegen"
-    schedule:
-      interval: "weekly"
+  - package-ecosystem: "gomod"\n    directory: "/"\n    schedule:\n      interval: "weekly"
```
Comment (bogdandrutu) [inline .github/dependabot.yml:4]: Haha :)) script does not work as expected on your machine. What do you use?

Do you mind reverting this change, or fix the script?

Comment (aabmass) [reply .github/dependabot.yml]: GNU Make 4.3 on gLinux (Debian) :)

How does that look now?

Comment (bogdandrutu) [reply .github/dependabot.yml]: Much better :) Thank you

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**